### PR TITLE
Fix test_route_flap case

### PR DESCRIPTION
**Pull request by AndriiLozovyi**
Signed-off-by: Andrii-Yosafat Lozovyi <andrii-yosafatx.lozovyi@intel.com>

### Description of PR

Summary: This PR should fix test_route_flap, with such issues as:
1) It seems that TC didn't perform withdraw or announce operations properly
**Logs from exabgp:**
```
Mon, 06 Jun 2022 07:32:48 | INFO     | 539    | processes     | Command from process http-api : neighbor 10.1.0.32 withdraw route 192.182.216.0 next-hop 10.0.0.57 
Mon, 06 Jun 2022 07:32:48 | WARNING  | 539    | reactor       | no neighbor matching the command : withdraw route 192.182.216.0 next-hop 10.0.0.57

Mon, 06 Jun 2022 07:34:05 | INFO     | 539    | processes     | Command from process http-api : neighbor 10.1.0.32 announce route 192.182.216.0 next-hop 10.0.0.57 
Mon, 06 Jun 2022 07:34:05 | WARNING  | 539    | reactor       | no neighbor matching the command : announce route 192.182.216.0 next-hop 10.0.0.57
```
2) TC failed with t1 topo
```
dev_port = 'Ethernet0'
def get_ptf_send_ports(duthost, tbinfo, dev_port):
mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
> member_port = mg_facts['minigraph_portchannels'][dev_port]['members']
E KeyError: u'Ethernet0'
```

Even when cmd to announce or withdraw routes is fixed, we would end up with testbed being left with non default routes added:
**Withdrawn**
```
Mon, 06 Jun 2022 07:44:29 | INFO     | 539    | processes     | Command from process http-api : withdraw route 192.182.216.0 next-hop 10.10.246.254 
Mon, 06 Jun 2022 07:44:29 | INFO     | 539    | reactor       | Route removed : 192.182.216.0/32 next-hop 10.10.246.254
Mon, 06 Jun 2022 07:44:30 | INFO     | 539    | reactor       | Performing dynamic route update
Mon, 06 Jun 2022 07:44:30 | INFO     | 539    | reactor       | Updated peers dynamic routes successfully

$ vtysh -c 'show ip route 192.182.216.0'
Routing entry for 192.182.216.0/25
  Known via "bgp", distance 20, metric 0, best
  Last update 00:38:03 ago
  * 10.0.0.57, via PortChannel101, weight 1
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
```
**Announce**

```
Mon, 06 Jun 2022 07:47:00 | INFO     | 539    | processes     | Command from process http-api : announce route 192.182.216.0 next-hop 10.10.246.254 
Mon, 06 Jun 2022 07:47:00 | INFO     | 539    | reactor       | Route added to neighbor 10.10.246.29 local-ip 10.10.246.254 local-as 64600 peer-as 64600 router-id 10.10.246.254 family-allowed in-open : 192.182.216.0/32 next-hop 10.10.246.254
Mon, 06 Jun 2022 07:47:01 | INFO     | 539    | reactor       | Performing dynamic route update
Mon, 06 Jun 2022 07:47:01 | INFO     | 539    | reactor       | Updated peers dynamic routes successfully

$ vtysh -c 'show ip route 192.182.216.0'
Routing entry for 192.182.216.0/32
  Known via "bgp", distance 20, metric 0, best
  Last update 00:00:33 ago
  * 10.0.0.57, via PortChannel101, weight 1

$ vtysh -c 'show ip route 192.182.216.0/25'
Routing entry for 192.182.216.0/25
  Known via "bgp", distance 20, metric 0, best
  Last update 00:39:17 ago
  * 10.0.0.57, via PortChannel101, weight 1
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
```

Updated the logic of TC, and added aspath to cmd that withdraw or announce routes to bgp neighbour, also added teardown that announce default routes at end of TC:
**Withdrawn**
```
Mon, 06 Jun 2022 09:01:33 | INFO     | 542    | processes     | Command from process http-api : withdraw route 192.182.216.0/25 next-hop 10.10.246.254 as-path [ 65534 64629 65511 ] 
Mon, 06 Jun 2022 09:01:33 | INFO     | 542    | reactor       | Route removed : 192.182.216.0/25 next-hop 10.10.246.254 as-path [ 65534 64629 65511 ]
Mon, 06 Jun 2022 09:01:34 | INFO     | 542    | reactor       | Performing dynamic route update
Mon, 06 Jun 2022 09:01:34 | INFO     | 542    | reactor       | Updated peers dynamic routes successfully

$ vtysh -c 'show ip route 192.182.216.0/25'
Routing entry for 192.182.216.0/25
  Known via "bgp", distance 20, metric 0, best
  Last update 00:00:39 ago
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
```

**Announce**
```
Mon, 06 Jun 2022 09:02:31 | INFO     | 542    | processes     | Command from process http-api : announce route 192.182.216.0/25 next-hop 10.10.246.254 as-path [ 65534 64629 65511 ] 
Mon, 06 Jun 2022 09:02:31 | INFO     | 542    | reactor       | Route added to neighbor 10.10.246.29 local-ip 10.10.246.254 local-as 64600 peer-as 64600 router-id 10.10.246.254 family-allowed in-open : 192.182.216.0/25 next-hop 10.10.246.254 as-path [ 65534 64629 65511 ]
Mon, 06 Jun 2022 09:02:32 | INFO     | 542    | reactor       | Performing dynamic route update
Mon, 06 Jun 2022 09:02:32 | INFO     | 542    | reactor       | Updated peers dynamic routes successfully

$ vtysh -c 'show ip route 192.182.216.0/25'
Routing entry for 192.182.216.0/25
  Known via "bgp", distance 20, metric 0, best
  Last update 00:00:30 ago
  * 10.0.0.57, via PortChannel101, weight 1
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
```
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Fix issue when TC was executed on t1 topo and fix withdrawn and announce operations.
#### How did you verify/test it?
Run test case on t0, t1, t1-lag topology:
`route/test_route_flap.py::test_route_flap PASSED`
#### Any platform specific information?
```
SONiC Software Version: SONiC.master.105614-dirty-20220602.215204
Distribution: Debian 11.3
Kernel: 5.10.0-12-2-amd64
Build commit: 0552d6b17
Build date: Thu Jun  2 21:58:41 UTC 2022
Built by: AzDevOps@sonic-build-workers-001KKZ
Platform: x86_64-accton_as9516_32d-r0
HwSKU: newport
```

### CODE REVIEW

**Comment by yutongzhang-microsoft** [comment]
@AndriiLozovyi I wonder why the TC will end up with testbed being left with non default routes added. In this TC, we don't have the operation to withdraw whole routes. Can you try on a SONiC Software Version without `dirty`?

File: tests/route/test_route_flap.py [hunk 1]
```diff
@@ -61,12 +74,26 @@ def get_ptf_recv_ports(duthost, tbinfo):
...
 
 
+def check_route(duthost, route, dev_port, operation):
+    out = json.loads(duthost.shell('vtysh -c "show ip route {} json"'.format(route), verbose=False)['stdout'])
+    result = [nexthop['interfaceName'] for nexthop in out[route][0]['nexthops']]
+    if operation == WITHDRAW:
+        pytest_assert(dev_port not in result, "Route {} was not withdraw {}".format(route, result))
+    else:
+        pytest_assert(dev_port in result, "Route {} was not announced {}".format(route, result))
+
```
Comment (yutongzhang-microsoft) [inline tests/route/test_route_flap.py:95]: This may take some time from ASIC sync with sonic, I think using `wait_until` is better here ?

**Comment by AndriiLozovyi** [comment]
> @AndriiLozovyi I wonder why the TC will end up with testbed being left with non default routes added. In this TC, we don't have the operation to withdraw whole routes. Can you try on a SONiC Software Version without `dirty`?

By default TC loops trough routes that are announced on DUT, and first it will try to withdraw route without specifying a prefix, which will lead to removing a route with **/32 prefix** which doesn't exits. 

After this TC will try to announce this route, and again without specifying a prefix, it will announce a route with **/32 prefix**, which didn't exist before TC run on DUT, and will leave DUT in such state, this will happen with all routes that are used by test case. 

Example:
```
Before
$ vtysh -c 'show ip route 192.182.216.0'
Routing entry for 192.182.216.0/25
  Known via "bgp", distance 20, metric 0, best
  Last update 00:38:03 ago
  * 10.0.0.57, via PortChannel101, weight 1
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
  
After
$ vtysh -c 'show ip route 192.182.216.0'
Routing entry for 192.182.216.0/32
  Known via "bgp", distance 20, metric 0, best
  Last update 00:00:33 ago
  * 10.0.0.57, via PortChannel101, weight 1

$ vtysh -c 'show ip route 192.182.216.0/25'
Routing entry for 192.182.216.0/25
  Known via "bgp", distance 20, metric 0, best
  Last update 00:39:17 ago
  * 10.0.0.57, via PortChannel101, weight 1
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
```

File: tests/route/test_route_flap.py (hunk 1 above)
Comment (AndriiLozovyi) [reply tests/route/test_route_flap.py:95]: @yutongzhang-microsoft, 
I used sleep for 1 second before calling this functions, and it looked like it was enough.

**Review by yutongzhang-microsoft — APPROVED** [review]

**Review by ZhaohuiS — APPROVED** [review]
LGTM

File: tests/route/test_route_flap.py [hunk 2]
```diff
@@ -94,40 +121,41 @@ def send_recv_ping_packet(ptfadapter, ptf_send_port, ptf_recv_ports, dst_mac, sr
...
-                for key, value in nexthops_info[0].items():
-                    if key == 'ip':
-                        nexthop = value
-                    if key == 'interfaceName':
-                        dev_port = value
+            # Use only multipath routes, othervise there will be announced new routes to T0 neigbours on t1 topo
+            multipath = iproute_info[route_prefix][0].get('multipath', False)
+            if multipath:
+                out = iproute_info[route_prefix][0].get('path').split(' ')
+                aspath = out[1:]
```
Comment (wenyiz2021) [inline tests/route/test_route_flap.py:152]: hello, how do you decide aspath here?
e.g. u'path': u'65200 64931 65900'
aspath would be `[u'64931', u'65900']`, but why?

**Pull request merged**